src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -419,6 +422,13 @@ private void updateEventTriggeredFleetStatusData(DeploymentInformation deploymen
             removedDependenciesSet.forEach(serviceFssTracksMap::remove);
             removedDependenciesSet.clear();
 
+            // Do not send empty FSS update when network reconnects in order to avoid spamming messages on flaky
+            // network.
+            // TODO: throttling mechanism for FSS updates
+            if (updatedGreengrassServiceSet.isEmpty() && Trigger.RECONNECT.equals(trigger)) {
```

Comment:
this optimizes for the worst case condition of a flaky network (or two devices with the same name booting each other off repeatedly). However there is a nice thing that customers lose here - when there is a network outage, they could see when GG reconnects. 

Rather than dropping all, did you consider keeping track of the last time a message was sent and not sending if it was within a time window, like 60s or 5m?

That would eliminate the spam, but still provide some more value.

If reconnect trigger is intended only to be used when something is removed while the device is offline, then update the comments to add more detail about that.

Reply:
Added a 60 seconds time window